tests/limits/test_vif_limit.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+
+import ipaddress
+import logging
+import os
+import tempfile
+
+from pkgfixtures import host_with_saved_yum_state
+
+# Requirements:
+# - one XCP-ng host (--host) >= 8.2
+# - a VM (--vm)
+# - the first network on the host can be used to reach the host
+
+vif_limit = 16
+interface_name = "enX"
+vcpus = '8'
+
+# There is a ResourceWarning due to background=True on an ssh call
+# We do ensure the processes are killed
+@pytest.mark.filterwarnings("ignore::ResourceWarning")
+@pytest.mark.debian_uefi_vm
+class TestVIFLimit:
+    def test_vif_limit(self, host_with_saved_yum_state, imported_vm):
+        host = host_with_saved_yum_state
+        vm = imported_vm
+        if (vm.is_running()):
+            logging.info("VM already running, shutting it down first")
+            vm.shutdown(verify=True)
+
+        network_uuid = vm.vifs()[0].param_get('network-uuid')
+        existing_vifs = len(vm.vifs())
+
+        logging.info(f'Get {vcpus} vCPUs for the VM')
+        original_vcpus_max = vm.param_get('VCPUs-max')
+        original_vcpus_at_startup = vm.param_get('VCPUs-at-startup')
+        vm.param_set('VCPUs-max', vcpus)
+        vm.param_set('VCPUs-at-startup', vcpus)
+
+        logging.info('Create VIFs before starting the VM')
+        vifs = []
+        for i in range(existing_vifs, vif_limit):
+            vif = vm.create_vif(i, network_uuid=network_uuid)
+            vifs.append(vif)
```

Comment:
VIF creation is not rolled back if subsequent code between this and the benchmark code fails (e.g. failed to install iperf3).